container-registry/container-analysis/sample.rb

```diff
@@ -0,0 +1,346 @@
+  formatted_project = Grafeas::V1::GrafeasClient.project_path occurrence_project
+
+  occurrence = {
+    note_name:     formatted_note,
+    resource_uri:  resource_url,
+    vulnerability: {
+      package_issue: [
+        affected_cpe_uri:     "your-uri-here:",
+        affected_package:     "your-package-here",
+        min_affected_version: { kind: :MINIMUM },
+        fixed_version:        { kind: :MAXIMUM }
+      ]
+    }
+  }
+
+  response = client.create_occurrence formatted_project, occurrence
+  puts response.name
+  # [END containeranalysis_create_occurrence]
+  response
+end
+
+def delete_occurrence occurrence_id:, project_id:
+  # [START containeranalysis_delete_occurrence]
+  # occurrence_id = "The API-generated identifier associated with the
+  #                  occurrence"
+  # project_id    = "The Google Cloud project ID of the occurrence to delete"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+
+  formatted_parent = Grafeas::V1::GrafeasClient.occurrence_path project_id, occurrence_id
+  client.delete_occurrence formatted_parent
+  # [END containeranalysis_delete_occurrence]
+end
+
+def get_note note_id:, project_id:
+  # [START containeranalysis_get_note]
+  # note_id    = "The identifier for the note to retrieve"
+  # project_id = "The Google Cloud project ID of the note to retrieve"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+
+  formatted_note = Grafeas::V1::GrafeasClient.note_path project_id, note_id
+  response = client.get_note formatted_note
+  puts response.name
+  # [END containeranalysis_get_note]
+  response
+end
+
+def get_occurrence occurrence_id:, project_id:
+  # [START containeranalysis_get_occurrence]
+  # occurrence_id = "The API-generated identifier associated with the
+  #                  occurrence"
+  # project_id    = "The Google Cloud project ID of the occurrence to retrieve"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+
+  formatted_parent = Grafeas::V1::GrafeasClient.occurrence_path project_id, occurrence_id
+  response = client.get_occurrence formatted_parent
+  puts response.name
+  # [END containeranalysis_get_occurrence]
+  response
+end
+
+def get_occurrences_for_image resource_url:, project_id:
+  # [START containeranalysis_occurrences_for_image]
+  # Initialize the client
+  # resource_url = "The URL of the resource associated with the occurrence.
+  #                 eg. https://gcr.io/project/image@sha256:123"
+  # project_id    = "The Google Cloud project ID of the occurrences to
+  #                  retrieve"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+
+  formatted_parent = Grafeas::V1::GrafeasClient.project_path project_id
+  filter = "resourceUrl = \"#{resource_url}\""
+  count = 0
+  client.list_occurrences(formatted_parent, filter: filter).each do |occurrence|
+    # Process occurrence here
+    puts occurrence
+    count += 1
```

Comment:
Using `count = 0` and then `count += 1` seems like it could be improved.

Could we have something like:
```rb
occurrences = client.list_occurrences(formatted_parent, filter: filter)
occurrences.each {|occurrence| puts occurrence }
puts "Found #{occurrences.count} occurrences"
```

Here and other places.

Reply:
The point of this sample is to show how you would retrieve a list of occurrences and do some generic work on each one. Getting the count of the results isn't actually necessary in itself, just the iteration part.

We could completely remove the counting aspect, but I think it's useful to have that part consistent across all languages

Reply:
OK, but can we just use `each_with_index` then instead? We can still have a  multiline for-loop body (maybe with a comment) if we expect some generic work to be done.

Reply:
What is the benefit of using `each_with_index` rather than `each`?

Reply:
So you don't need to define `count` in the loop body. Do you think it's cleaner?

Reply:
hmm I don't think so, I feel like using the index in that way would just complicate what would otherwise be a simple example. Is this pretty unconventional for Ruby?

I'm not completely opposed to removing the whole "count" part of the sample entirely and just printing the occurrence in each loop, but then it would be inconsistent with other languages